ic-agent/src/identity/basic.rs

```diff
@@ -28,15 +30,56 @@ impl BasicIdentity {
         let bytes: Vec<u8> = pem_reader
             .bytes()
             .collect::<Result<Vec<u8>, std::io::Error>>()?;
+        let file = std::str::from_utf8(&bytes).unwrap();
+        let pem_file: PrivateKeyDocument = file.parse().map_err(|_| PemError::PemError)?;
+        let key_info = pem_file.private_key_info();
+
+        // Check OID
+        if key_info.algorithm.oid != "1.3.101.112".parse().unwrap() {
+            return Err(PemError::KeyRejected(format!(
+                "Wrong OID, expected \"1.3.101.112\", got {}",
+                key_info.algorithm.oid,
+            )));
+        }
+
+        // Check that there are no parameters on the key
+        if key_info.algorithm.parameters.is_some() {
+            return Err(PemError::KeyRejected(
+                "Parameters on the key are not allowed".to_string(),
+            ));
+        }
+
+        // Retrieve the secret key and check that we have an uncompressed
```

Comment:
Note that the secret key is not an `uncompressed point` but it is a 32 byte string.

Reply:
Done